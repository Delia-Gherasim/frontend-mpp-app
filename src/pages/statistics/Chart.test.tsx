// Mock ResizeObserver
global.ResizeObserver = class {
    observe() {}
    unobserve() {}
    disconnect() {}
};

// Your test code
import {render} from '@testing-library/react';
import {BrowserRouter as Router} from 'react-router-dom';
import {describe, expect, it} from 'vitest';
import ChartPage from './Chart';

describe('ChartPage', () => {
    it('should render the Chart Page component', () => {
        const {container} = render(
            <Router>
                <ChartPage />
            </Router>,
        );
        const snapshot = container.innerHTML;
        expect(snapshot).toMatchSnapshot();
    });
});
