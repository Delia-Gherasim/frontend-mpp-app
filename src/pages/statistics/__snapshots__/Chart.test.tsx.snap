// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChartPage > should render the Chart Page component 1`] = `"<div class="container "><article class="article-header"><header><h2 class="title">Awesome App</h2></header></article><div class="sidebar-container "><div class="content"><hr><br><a class="sidebar-item" style="color: rgb(230, 230, 230);" href="/">See Database</a><hr><a class="sidebar-item" style="color: rgb(230, 230, 230);" href="/search">Search</a><hr><a class="sidebar-item" style="color: rgb(230, 230, 230);" href="/sort">Sort</a><hr><a class="sidebar-item" style="color: rgb(230, 230, 230);" href="/chart">Chart</a><hr></div></div><button class="hide-button" style="background-color: rgb(218, 165, 32); border-radius: 5px; color: black;">â˜°</button><section class="section-content"><h2 style="text-align: center;">Statistics</h2><div style="display: flex; justify-content: center; flex-direction: row; align-items: center;"><div style="display: flex; justify-content: center; flex-direction: column; align-items: center;"><h3 style="font-size: 14px; margin-bottom: 16px;">Percentages of Each Category</h3><div class="recharts-responsive-container" style="width: 300px; height: 300px; min-width: 0;"><div class="recharts-wrapper" style="position: relative; cursor: default; width: 100%; height: 100%; max-height: 300px; max-width: 300px;" role="region"><svg class="recharts-surface" width="300" height="300" style="width: 100%; height: 100%;" viewBox="0 0 300 300"><title></title><desc></desc><defs><clipPath id="recharts1-clip"><rect x="65" y="5" height="260" width="230"></rect></clipPath></defs><g class="recharts-cartesian-grid"><g class="recharts-cartesian-grid-horizontal"><line stroke-dasharray="3 3" stroke="#ccc" fill="none" x="65" y="5" width="230" height="260" x1="65" y1="5" x2="295" y2="5"></line><line stroke-dasharray="3 3" stroke="#ccc" fill="none" x="65" y="5" width="230" height="260" x1="65" y1="265" x2="295" y2="265"></line></g><g class="recharts-cartesian-grid-vertical"><line stroke-dasharray="3 3" stroke="#ccc" fill="none" x="65" y="5" width="230" height="260" x1="65" y1="5" x2="65" y2="265"></line><line stroke-dasharray="3 3" stroke="#ccc" fill="none" x="65" y="5" width="230" height="260" x1="295" y1="5" x2="295" y2="265"></line></g></g></svg><div tabindex="-1" class="recharts-tooltip-wrapper" style="visibility: hidden; pointer-events: none; position: absolute; top: 0px; left: 0px;"><div class="recharts-default-tooltip" style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"><p class="recharts-tooltip-label" style="margin: 0px;"></p></div></div></div></div></div><div style="display: flex; justify-content: center; flex-direction: column; align-items: center;"><h3 style="font-size: 14px; margin-bottom: 16px;">Percentages of Each Type of Electronics</h3><div class="recharts-responsive-container" style="width: 300px; height: 300px; min-width: 0;"><div class="recharts-wrapper" style="position: relative; cursor: default; width: 100%; height: 100%; max-height: 300px; max-width: 300px;" role="region"><svg class="recharts-surface" width="300" height="300" style="width: 100%; height: 100%;" viewBox="0 0 300 300"><title></title><desc></desc><defs><clipPath id="recharts4-clip"><rect x="65" y="5" height="260" width="230"></rect></clipPath></defs><g class="recharts-cartesian-grid"><g class="recharts-cartesian-grid-horizontal"><line stroke-dasharray="3 3" stroke="#ccc" fill="none" x="65" y="5" width="230" height="260" x1="65" y1="5" x2="295" y2="5"></line><line stroke-dasharray="3 3" stroke="#ccc" fill="none" x="65" y="5" width="230" height="260" x1="65" y1="265" x2="295" y2="265"></line></g><g class="recharts-cartesian-grid-vertical"><line stroke-dasharray="3 3" stroke="#ccc" fill="none" x="65" y="5" width="230" height="260" x1="65" y1="5" x2="65" y2="265"></line><line stroke-dasharray="3 3" stroke="#ccc" fill="none" x="65" y="5" width="230" height="260" x1="295" y1="5" x2="295" y2="265"></line></g></g></svg><div tabindex="-1" class="recharts-tooltip-wrapper" style="visibility: hidden; pointer-events: none; position: absolute; top: 0px; left: 0px;"><div class="recharts-default-tooltip" style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"><p class="recharts-tooltip-label" style="margin: 0px;"></p></div></div></div></div></div><div style="display: flex; justify-content: center; flex-direction: column; align-items: center;"><h3 style="font-size: 14px; margin-bottom: 16px;">Percentages of Each Type of Appliances</h3><div class="recharts-responsive-container" style="width: 300px; height: 300px; min-width: 0;"><div class="recharts-wrapper" style="position: relative; cursor: default; width: 100%; height: 100%; max-height: 300px; max-width: 300px;" role="region"><svg class="recharts-surface" width="300" height="300" style="width: 100%; height: 100%;" viewBox="0 0 300 300"><title></title><desc></desc><defs><clipPath id="recharts7-clip"><rect x="65" y="5" height="260" width="230"></rect></clipPath></defs><g class="recharts-cartesian-grid"><g class="recharts-cartesian-grid-horizontal"><line stroke-dasharray="3 3" stroke="#ccc" fill="none" x="65" y="5" width="230" height="260" x1="65" y1="5" x2="295" y2="5"></line><line stroke-dasharray="3 3" stroke="#ccc" fill="none" x="65" y="5" width="230" height="260" x1="65" y1="265" x2="295" y2="265"></line></g><g class="recharts-cartesian-grid-vertical"><line stroke-dasharray="3 3" stroke="#ccc" fill="none" x="65" y="5" width="230" height="260" x1="65" y1="5" x2="65" y2="265"></line><line stroke-dasharray="3 3" stroke="#ccc" fill="none" x="65" y="5" width="230" height="260" x1="295" y1="5" x2="295" y2="265"></line></g></g></svg><div tabindex="-1" class="recharts-tooltip-wrapper" style="visibility: hidden; pointer-events: none; position: absolute; top: 0px; left: 0px;"><div class="recharts-default-tooltip" style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"><p class="recharts-tooltip-label" style="margin: 0px;"></p></div></div></div></div></div></div></section></div>"`;
